dsl/cache/cache.go

```diff
@@ -0,0 +1,139 @@
+package cache
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var cache = map[string][]byte{}
+var metadata = map[string]CacheMetadata{}
+var accessCount uint = 0
+
+func init() {
+	loadConfigs()
+}
+
+// we use absolute file path to remove the potential of redundant loading of files
+// for example a user could load 'D:proj/json_load.json' and 'json_load.json', both containt the same
+// file but the cache key will be different.
+//
+//	for both reading and writing to the cache.
+func FetchFileBytes(file_path string) ([]byte, error) {
+	defer sweep()
+
+	abs_file_path, err := filepath.Abs(file_path)
```

Comment:
The returned path is not ensured to be unique
```suggestion
	if !filepath.isAbs(file_path) {
	    log.panic("path is not absolute")
	}
```